ui/src/app/views/project/show/variable/variable.list.component.ts

```diff
@@ -16,6 +18,24 @@ import {finalize, first} from 'rxjs/operators';
 export class ProjectVariablesComponent implements OnInit {
 
     @Input() project: Project;
+    @Input('warnings')
+    set warnings(data: Array<Warning>) {
+        if (data) {
+            this.variableWarning = new Map<string, Warning>();
+            this.unusedVariableWarning = new Array<Warning>();
+            data.forEach(v => {
+                let w = cloneDeep(v);
```

Comment:
same here. Check if you really need to make a cloneDeep

Reply:
I do not want to modify the warning received, because same ref is shared by several screen